dot/sync/warp_sync.go

```diff
@@ -0,0 +1,238 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package sync
+
+import (
+	"slices"
+	"time"
+
+	"github.com/ChainSafe/gossamer/dot/network"
+	"github.com/ChainSafe/gossamer/dot/network/messages"
+	"github.com/ChainSafe/gossamer/dot/peerset"
+	"github.com/ChainSafe/gossamer/dot/types"
+	consensus_grandpa "github.com/ChainSafe/gossamer/internal/primitives/consensus/grandpa"
+	primitives "github.com/ChainSafe/gossamer/internal/primitives/consensus/grandpa"
+	"github.com/ChainSafe/gossamer/lib/grandpa"
+	"github.com/libp2p/go-libp2p/core/peer"
+)
+
+type WarpSyncPhase uint
+
+const (
+	WarpProof = iota
+	TargetBlock
+	Completed
+)
+
+type WarpSyncStrategy struct {
+	// Strategy dependencies and config
+	peers            *peerViewSet
```

Comment:
the peer view set is important for `FullSync`, I am wondering if when WarpSync finishes the peerViewSet is lost or is it transferred to FullSync in some way?

Reply:
I'm working on some changes in the `service.go` to handle the strategy switch and I'll consider this there! thks!

Reply:
Have you found a way to pass those infos to the fullsync strategy?

Reply:
Yes, if you check at the [service](https://github.com/ChainSafe/gossamer/pull/4275/files#diff-99bfbc3a2947ae495ea0c0fd19bcdcea2fa24a9d7fec49b4a9237361bb6709eaR522) we are sharing the same `peerViewSet` reference for all strategies